src/test/java/io/confluent/kafkarest/integration/SimpleConsumerBinaryTest.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest.integration;
+
+import io.confluent.kafkarest.Errors;
+import io.confluent.kafkarest.Versions;
+import io.confluent.kafkarest.entities.BinaryConsumerRecord;
+import jersey.repackaged.com.google.common.collect.ImmutableMap;
+import kafka.utils.TestUtils;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.Before;
+import org.junit.Test;
+import scala.collection.JavaConversions;
+
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+import static io.confluent.kafkarest.TestUtils.assertErrorResponse;
+
+public class SimpleConsumerBinaryTest extends AbstractConsumerTest {
+
+  private static final String topicName = "topic1";
+
+  private final List<ProducerRecord<byte[], byte[]>> recordsOnlyValues = Arrays.asList(
+      new ProducerRecord<byte[], byte[]>(topicName, "value".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "value2".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "value3".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "value4".getBytes())
+  );
+
+  private final List<ProducerRecord<byte[], byte[]>> recordsWithKeys = Arrays.asList(
+      new ProducerRecord<byte[], byte[]>(topicName, "key".getBytes(), "value".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "key".getBytes(), "value2".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "key".getBytes(), "value3".getBytes()),
+      new ProducerRecord<byte[], byte[]>(topicName, "key".getBytes(), "value4".getBytes())
+  );
+
+  private static final GenericType<List<BinaryConsumerRecord>> binaryConsumerRecordType
+      = new GenericType<List<BinaryConsumerRecord>>() {
+  };
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    final int numPartitions = 1;
```

Comment:
Do you think there's any value in testing multiple partitions or is that simple enough in terms of passing the parameters through that it's not worth it?